raster-foundry/raster-foundry
# Make RasterSource Configurable via Environment

## Overview

Adds ability to choose raster source type based on environment. This is to allow easier testing between the GeoTiffRasterSource and the GDALRasterSource

### Checklist


### Demo
```
backsplash_1        |   _   _   _        _ _
backsplash_1        |  | |_| |_| |_ _ __| | | ___
backsplash_1        |  | ' \  _|  _| '_ \_  _(_-<
backsplash_1        |  |_||_\__|\__| .__/ |_|/__/
backsplash_1        |              |_|
backsplash_1        | 13:32:37.109 [http4s-0] INFO  org.http4s.server.blaze.BlazeServerBuilder - http4s v0.20.0-M4 on blaze v0.14.0-M11 started at http://0.0.0.0:8080/
backsplash_1        | 13:32:46.879 [blaze-selector-0-1] INFO  o.http4s.blaze.channel.nio1.NIO1SocketServerGroup - Accepted connection from /172.27.0.10:55806
backsplash_1        | 13:32:47.886 [http4s-0] DEBUG com.rasterfoundry.backsplash.server.Authenticators - Authentication - Getting User auth0|59318a9d2fbbca3e16bcfc92 from DB
backsplash_1        | 13:32:48.576 [http4s-0] DEBUG com.rasterfoundry.backsplash.server.Authorizers - Checking Project Auth User: auth0|59318a9d2fbbca3e16bcfc92 => Project: 6477b59f-2066-4b4c-8e0b-6232fbd82b3e with DB
backsplash_1        | 13:32:49.056 [db-client-8] DEBUG com.rasterfoundry.backsplash.MosaicImplicits - Reading Histogram 7acc48c0-ccfc-435b-962d-a3355e139791 10-163-395
backsplash_1        | 13:32:49.060 [http4s-1] DEBUG com.rasterfoundry.backsplash.MosaicImplicits - Retrieving Histograms for 7acc48c0-ccfc-435b-962d-a3355e139791 from Source
backsplash_1        | 13:32:49.060 [http4s-2] DEBUG com.rasterfoundry.backsplash.MosaicImplicits - Reading Tile 7acc48c0-ccfc-435b-962d-a3355e139791 10-163-395
backsplash_1        | 13:32:49.067 [http4s-2] DEBUG com.rasterfoundry.backsplash.BacksplashImage - Reading 10-163-395 - Image: 7acc48c0-ccfc-435b-962d-a3355e139791 at s3://rasterfoundry-development-data-us-east-1/user-uploads/auth0%7C59318a9d2fbbca3e16bcfc92/4b30cef0-aea0-4c6b-9bc5-b726a56e8e51/PSScene4Band-20180829_182546_1025-analytic_sr.tif
backsplash_1        | 13:32:49.069 [http4s-2] DEBUG com.rasterfoundry.backsplash.BacksplashImage$ - Using GDAL Raster Source: s3://rasterfoundry-development-data-us-east-1/user-uploads/auth0%7C59318a9d2fbbca3e16bcfc92/4b30cef0-aea0-4c6b-9bc5-b726a56e8e51/PSScene4Band-20180829_182546_1025-analytic_sr.tif
```

## Testing Instructions

 * Modify `docker.java.yml` and set logging level to `DEBUG` and set environment to use `GDAL`
 * Assemble and start application, verify via logging output when using application that GDAL is being used
